client/src/test/java/org/eclipse/hono/client/impl/HonoClientImplTest.java

```diff
@@ -479,6 +479,26 @@ public void testConnectFailsAfterShutdown(final TestContext ctx) {
                 }));
     }
 
+    /**
+     * Verifies that if a client disconnects from the server, then an attempt to connect again will be succcessful.
+     *
+     * @param ctx The test execution context.
+     */
+    @Test
+    public void testConnectSucceedsAfterDisconnect(final TestContext ctx) {
+
+        // If a client disconnects from the server
+        Async disconnect = ctx.async();
+        client.disconnect();
```

Comment:
this needs to be the variant with a handler so that we know, when it's time to reconnect ...